Add test case for `storage/blob/s3/writer/option` (#656)
vdaas/vald: docs/contributing/coding-style.md

You can refer to [this section](#Struct-initialization) for more details of the use case of this pattern.

We strongly recommend the following implementation to set the value using functional option.

> If set to an invalid value, it should return the error defined in the [internal/errors/option.go](https://github.com/vdaas/vald/blob/master/internal/errors/option.go).

Reviewer: ```suggestion
If the invalid value is set to the functional option, the `ErrInvalidOption` error defined in the [internal/errors/option.go](https://github.com/vdaas/vald/blob/master/internal/errors/option.go) should be returned.
```

> If the invalid value is set to the functional option, the `ErrInvalidOption` error defined in the [internal/errors/option.go](https://github.com/vdaas/vald/blob/master/internal/errors/option.go) should be returned.

Author: ```suggestion

If the invalid value is set to the functional option, the `ErrInvalidOption` error defined in the [internal/errors/option.go](https://github.com/vdaas/vald/blob/master/internal/errors/option.go) should be returned.
```

```go
func WithVersion(version string) Option {
    return func(c *client) error {
        if len(version) == 0 {

>             return errors.ErrEmptyString

Author: ```suggestion
            return errors.ErrInvalidOption("version", version)
```

We strongly recommend the following implementation to set the value using functional option.

> If the invalid value is set to the functional option, the `ErrInvalidOption` error defined in the [internal/errors/option.go](https://github.com/vdaas/vald/blob/master/internal/errors/option.go) should be returned.

Author: ```suggestion
If an invalid value is set to the functional option, the `ErrInvalidOption` error defined in the [internal/errors/option.go](https://github.com/vdaas/vald/blob/master/internal/errors/option.go) should be returned.

The name argument (the first argument) of the `ErrInvalidOption` error should be the same as the functional option name without the `With` prefix. 

For example, the functional option name `WithVersion` should return the error with the argument `name` as `version`.

```

We strongly recommend the following implementation to set the value using functional option.

If an invalid value is set to the functional option, the `ErrInvalidOption` error defined in the [internal/errors/option.go](https://github.com/vdaas/vald/blob/master/internal/errors/option.go) should be returned.

The name argument (the first argument) of the `ErrInvalidOption` error should be the same as the functional option name without the `With` prefix. 

> For example, the functional option name `WithVersion` should return the error with the argument `name` as `version`.

Reviewer 2: LGTM, one point, It would be better to write about option load loop error handling using `errors.ErrOptionFailed`

Author: Thank you for your review.
I think so too.
But regarding error handling, I think it is better to add it to the proposal's PR because there is a possibility that it will change when the proposal is applied. 🤔

Reviewer 2: I understand about it, but we should write current error handling use case first,  after proposal approved you should fix it

Reviewer 2: It's not so difficult.

Author: thank you for your opinion.
I will add it to this PR soon

func WithTimeout(dur string) Option {
    func(c *client) error {
        if dur == "" {

>             return errors.ErrEmptyString

Author: ```suggestion
            return errors.ErrInvalidOption("dur", dur)
```

func WithTimeout(dur string) Option {
    func(c *client) error {
        if dur == "" {

>             return errors.ErrInvalidOption("dur", dur)

Author: ```suggestion
            return errors.ErrInvalidOption("timeout", dur)
```

func WithHosts(hosts ...string) Option {
    return func(c *client) error {
        if len(hosts) == 0 {

>             return ErrEmptySlice

Author: ```suggestion
            return errors.ErrInvalidOption("hosts", hosts)
```
